src/OrchardCore/OrchardCore.Scripting.JavaScript/JavaScriptEngine.cs

```diff
@@ -19,7 +22,34 @@ public JavaScriptEngine(IMemoryCache memoryCache)
 
         public IScriptingScope CreateScope(IEnumerable<GlobalMethod> methods, IServiceProvider serviceProvider, IFileProvider fileProvider, string basePath)
         {
-            var engine = new Engine();
+            var engine = new Engine(options =>
+            {
+                // make JsonArray behave like JS array 
+                options.Interop.WrapObjectHandler = static (e, target, type) =>
+                {
+                    var wrapped = new ObjectWrapper(e, target);
+                    if (target is JsonArray)
+                    {
+                        wrapped.Prototype = e.Intrinsics.Array.PrototypeObject;
+                    }
+                    return wrapped;
+                };
+
+                // we cannot access this[string] with anything else than JsonObject, otherwise itw will throw 
```

Comment:
When you add comments, please start with a Capital letter and add a period at the end of the sentence.